doc/rst/technotes/libraries.rst

```diff
@@ -384,6 +416,69 @@ For example:
    Make sure your Python functionality is also complete before calling this
    function.
 
+.. note::
+
+  If you are taking advantage of the generated ``__init__.py`` initializer
+  file to import the output directory as a package, you do not need to call
+  ``chpl_cleanup()`` yourself because it is already registered to be called
+  at program exit. The generated initializer is explained below.
+
+.. _Python_Init_File:
+
+Python Init File
+----------------
+
+A simple ``__init__.py`` file is generated in the output directory along with
+the Python module. It looks roughly like the following:
+
+.. code-block:: python
+
+  import atexit
+
+  #
+  # Here directoryName is the name of the directory containing your Python
+  # module, and moduleName is the name of your Python module.
+  #
+  from directoryName.moduleName import *
+
+  atexit.register(moduleName.chpl_cleanup)
+
+The initializer file transforms the output directory into a Python package.
+It will also register ``chpl_cleanup()`` to be called automatically at
+program exit.
+
+The directory containing generated Python code must be visible to any script
+that attempts to import it as a package. One option is to place the directory
+somewhere in an existing codebase and use an absolute import. Another option
+is to place the package directory on your ``PYTHONPATH`` environment variable.
+
+.. code-block:: bash
+
+  # Builds Python module as foo/foo.py
+  chpl --library-python foo.chpl
+
+  # Adds the current directory to your PYTHONPATH
+  export PYTHONPATH="$PWD:$PYTHONPATH"
+
+From within your Python script:
+
+.. code-block:: Python
+
+  import foo
+
+  #
+  # Setup and use foo as normal. Note that we no longer have to call
+  # ``chpl_cleanup()`` when we are finished.
+  #
+  foo.chpl_setup()
+  foo.baz(2)
+
+.. note::
+
+  The Chapel compiler will not generate an initializer file if a file with
+  the name ``__init__.py`` already exists in the output directory. The
+  compiler will emit a warning instead.
```

Comment:
Is there a way to disable this warning for projects that intentionally utilize this behavior?

Reply:
Sounds like maybe I _should_ add a compiler flag?

Reply:
I would hold off on creating a one-off compiler flag for this warning control. There are more organized efforts that could enable this in the future:

- different categories of warnings (https://github.com/chapel-lang/chapel/issues/11576)
- generalized warning control (https://github.com/chapel-lang/chapel/issues/11576)